### PR TITLE
refactor: suberror

### PR DESCRIPTION
**Pull request by qazxcdswe123**

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -60,12 +60,33 @@ jobs:
           curl -fsSL https://cli.moonbitlang.com/install/unix.sh | bash
           echo "$HOME/.moon/bin" >> $GITHUB_PATH
 
+      - name: upgrade-core
```
Comment (qazxcdswe123) [inline .github/workflows/check.yml:63]: Do we really needs this though, it can be automatically solved after 1 day? (bump core)

Comment (Yoorkin) [reply .github/workflows/check.yml:63]: @bzy-debug

**Review by Yoorkin — APPROVED** [review]
LGTM

File: .github/workflows/check.yml (hunk 1 above)
Comment (bzy-debug) [reply .github/workflows/check.yml:63]: don't really need

**Comment by Yoorkin** [comment]
There are some new breaking changes that caused the CI to fail.

```
 [4018] Error: Type ParseError is not an error type
     ╭─[/Users/runner/work/x/x/json5/lex_string.mbt:101:15]
     │
 101 │       None => raise InvalidEof
     │               ────────┬───────  
     │                       ╰───────── Type ParseError is not an error type
─────╯
```

**Pull request merged**